wled00/util.cpp

```diff
@@ -422,74 +424,60 @@ um_data_t* simulateSound(uint8_t simulationId)
     um_data->u_data[7] = &binNum;
   } else {
     // get arrays from um_data
-    fftResult =  (uint8_t*)um_data->u_data[2];
+    fftResult = (uint8_t*)um_data->u_data[2];
   }
 
   uint32_t ms = millis();
 
   switch (simulationId) {
     default:
     case UMS_BeatSin:
-      for (int i = 0; i<16; i++)
-        fftResult[i] = beatsin8(120 / (i+1), 0, 255);
-        // fftResult[i] = (beatsin8(120, 0, 255) + (256/16 * i)) % 256;
-        volumeSmth = fftResult[8];
+      for (int i = 0; i<16; i++) fftResult[i] = beatsin8(120 / (i+1), 0, 255);
+      volumeSmth = fftResult[8];
       break;
     case UMS_WeWillRockYou:
       if (ms%2000 < 200) {
         volumeSmth = random8(255);
-        for (int i = 0; i<5; i++)
-          fftResult[i] = random8(255);
-      }
-      else if (ms%2000 < 400) {
+        for (int i = 0; i < 5; i++) fftResult[i] = random8(255);
+      } else if (ms%2000 < 400) {
```

Comment:
not related to this PR directly but something I recently learned:
since modulo is slow on ESP8266, ESP32 C3 and maybe even S2 it can be replaced with a simple `&` if done with a power of 2. If the 2000ms is not so important (which I am guessing it is not) all the `ms%2000`could be replaced by `ms&2047` or `ms&0x7F`